### PR TITLE
Period accepts datetime64 value

### LINKED ISSUE #9054: ENH: Period to accept datetime64 value?
I didn't directly find something about this, but is there a reason why the `Period(Index)` constructor does not take datetime64 values? As it does accept `datetime`:

```
In [15]: t = pd.Timestamp('2012-01-01')

In [16]: t.to_pydatetime()
Out[16]: datetime.datetime(2012, 1, 1, 0, 0)

In [17]: t.asm8
Out[17]: numpy.datetime64('2012-01-01T01:00:00.000000000+0100')

In [19]: pd.Period(t, freq='D')
Out[19]: Period('2012-01-01', 'D')

In [20]: pd.Period(t.to_pydatetime(), freq='D')
Out[20]: Period('2012-01-01', 'D')

In [21]: pd.Period(t.asm8, freq='D')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-21-f08079765cf6> in <module>()
----> 1 pd.Period(t.asm8, freq='D')

C:\Anaconda\lib\site-packages\pandas\tseries\period.pyc in __init__(self, value,
 freq, ordinal, year, month, quarter, day, hour, minute, second)
    141         else:
    142             msg = "Value must be Period, string, integer, or datetime"
--> 143             raise ValueError(msg)
    144
    145         base, mult = _gfc(freq)

ValueError: Value must be Period, string, integer, or datetime
```

### PR DESCRIPTION
**Pull request by jcrist**
Added support for `datetime64` value for Period. Fixes #9054.

### CODE REVIEW

File: pandas/tseries/tests/test_period.py [hunk 1]
```diff
@@ -226,7 +226,9 @@ def test_period_constructor(self):
 
         i1 = Period(date(2007, 1, 1), freq='M')
         i2 = Period(datetime(2007, 1, 1), freq='M')
+        i3 = Period(np.datetime64('2007-01-01'), freq='M')
```
Comment (jreback) [inline pandas/tseries/tests/test_period.py]: can you add a couple of more tests that have different numpy units, 

e.g. 

```
In [3]: np.datetime64('2007-01-01 00:00:00',dtype='M8[s]').dtype
Out[3]: dtype('<M8[s]')
```

**Comment by jreback** [comment]
pls add a release note as wel, looks good otherwise.

**Comment by cpcloud** [comment]
+1, will be awhile for travis, lots of PRs today :)

**Comment by jreback** [comment]
can you squash into a single commit. pls ping when done.

**Comment by jcrist** [comment]
@jreback : squashed.

**Comment by cpcloud** [comment]
closing in favor of #9896

**Pull request closed without merging**